jmxtrans-core/src/test/java/com/googlecode/jmxtrans/cluster/ClusterConnectTest.java

```diff
@@ -0,0 +1,197 @@
+            setupClusterClient();
+        }
+    }
+
+    @After
+    public void after() throws Exception {
+    }
+
+    @Ignore @Test
+    public void testInterProcessLock() throws Exception{
+        InterProcessMutex lock  = new InterProcessMutex(clClient, "/testcases/iplock" );
+        InterProcessMutex lock2  = new InterProcessMutex(clClient, "/testcases/iplock" );
+        lock.acquire(10, TimeUnit.SECONDS);
+
+        Thread.sleep(2000);
+        lock2.acquire(2,TimeUnit.SECONDS);
+        Thread.sleep(2000);
+
+        log.info("Lock 1 has to lock: " + lock.isAcquiredInThisProcess());
+        log.info("Lock 2 has to lock: " + lock2.isAcquiredInThisProcess());
+
+        Thread.sleep(120000);
+
+        log.info("Lock 1 has to lock: " + lock.isAcquiredInThisProcess());
+        log.info("Lock 2 has to lock: " + lock2.isAcquiredInThisProcess());
+    }
+
+    @Test
+    public void testZookeeperConnection() throws Exception {
+        try {
+            clClient.create().creatingParentContainersIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath("/jmxtrans/jvms/jvm_01/config", "demo config 1".getBytes());
+            clClient.create().creatingParentContainersIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath("/jmxtrans/jvms/jvm_01/affinity", ("worker_01").getBytes());
+            clClient.create().creatingParentContainersIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath("/jmxtrans/jvms/jvm_01/owner");
+
+            clClient.create().creatingParentContainersIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath("/jmxtrans/jvms/jvm_02/config", "demo config 2".getBytes());
+            clClient.create().creatingParentContainersIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath("/jmxtrans/jvms/jvm_02/affinity", ("worker_02").getBytes());
+            clClient.create().creatingParentContainersIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath("/jmxtrans/jvms/jvm_02/owner");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        ClusterConfigClient clusterConnect1 = new ClusterConfigClient();
+        clusterConnect1.initialize(clConfig.getString("zookeeper.connectionstring"),
+                "worker_01",
+                clConfig.getString("zookeeper.heartbeatpath"),
+                clConfig.getString("zookeeper.configpath"));
+
+        ClusterConfigClient clusterConnect2 = new ClusterConfigClient();
+        clusterConnect2.initialize(clConfig.getString("zookeeper.connectionstring"),
+                "worker_02",
+                clConfig.getString("zookeeper.heartbeatpath"),
+                clConfig.getString("zookeeper.configpath"));
+
+
+
+        clusterConnect1.start();
+        Thread.sleep(5000);
```

Comment:
Thread.sleep() in a test is a code smell. Ideally, you should have proper synchronisation and be able to wait() instead of sleep(), or at least do some frequent polling (have a look at [awaitility](https://github.com/jayway/awaitility)).